### ISSUE TITLE
(3.0.0rc02) caed exhausts file handles

### ISSUE THREAD

**Issue by deltecent** [ElvishArtisan/rivendell#439]
```
May 30 16:50:37 maxsrv rdairplay: finished event: Line: 131  Cart: 11572  Cut: 1 Card: 0  Stream: -1  Port: 0
May 30 16:50:37 maxsrv caed: Error: alsaLoadPlayback(/var/snd/045001_031.wav)   openWave() failed to open file
May 30 16:50:37 maxsrv caed: unable to allocate stream for card 0
May 30 16:50:37 maxsrv rdairplay: finished event: Line: 132  Cart: 45001  Cut: 31 Card: 0  Stream: -1  Port: 1
May 30 16:50:37 maxsrv caed: Error: alsaLoadPlayback(/var/snd/010666_001.wav)   openWave() failed to open file
May 30 16:50:37 maxsrv caed: unable to allocate stream for card 0
May 30 16:50:37 maxsrv rdairplay: finished event: Line: 133  Cart: 10666  Cut: 1 Card: 0  Stream: -1  Port: 0
May 30 16:50:37 maxsrv caed: Error: alsaLoadPlayback(/var/snd/010397_001.wav)   openWave() failed to open file
May 30 16:50:37 maxsrv caed: unable to allocate stream for card 0
May 30 16:50:37 maxsrv rdairplay: finished event: Line: 134  Cart: 10397  Cut: 1 Card: 0  Stream: -1  Port: 1
May 30 16:50:37 maxsrv caed: Error: alsaLoadPlayback(/var/snd/010729_001.wav)   openWave() failed to open file
May 30 16:50:37 maxsrv caed: unable to allocate stream for card 0
May 30 16:50:38 maxsrv rdairplay: finished event: Line: 135  Cart: 10729  Cut: 1 Card: 0  Stream: -1  Port: 0
May 30 16:50:38 maxsrv caed: Error: alsaLoadPlayback(/var/snd/060001_010.wav)   openWave() failed to open file
May 30 16:50:38 maxsrv caed: unable to allocate stream for card 0
May 30 16:50:38 maxsrv rdairplay: finished event: Line: 136  Cart: 60001  Cut: 10 Card: 0  Stream: -1  Port: 1
May 30 16:50:38 maxsrv caed: Error: alsaLoadPlayback(/var/snd/011582_001.wav)   openWave() failed to open file
May 30 16:50:38 maxsrv caed: unable to allocate stream for card 0
May 30 16:50:38 maxsrv rdairplay: finished event: Line: 137  Cart: 11582  Cut: 1 Card: 0  Stream: -1  Port: 0
May 30 16:50:38 maxsrv caed: Error: alsaLoadPlayback(/var/snd/045003_014.wav)   openWave() failed to open file
May 30 16:50:38 maxsrv caed: unable to allocate stream for card 0
May 30 16:50:38 maxsrv rdairplay: finished event: Line: 138  Cart: 45003  Cut: 14 Card: 0  Stream: -1  Port: 1
May 30 16:50:38 maxsrv caed: Error: alsaLoadPlayback(/var/snd/011653_001.wav)   openWave() failed to open file
May 30 16:50:38 maxsrv caed: unable to allocate stream for card 0
May 30 16:50:38 maxsrv rdairplay: finished event: Line: 139  Cart: 11653  Cut: 1 Card: 0  Stream: -1  Port: 0
May 30 16:50:38 maxsrv caed: Error: alsaLoadPlayback(/var/snd/070000_001.wav)   openWave() failed to open file
May 30 16:50:38 maxsrv caed: unable to allocate stream for card 0
```
The only way to fix is to ```systemctl restart rivendell```.

This happens randomly. I'm still obtaining more information.

**Comment by deltecent** [comment]
It looks like ```caed``` may not be closing file handles, and we're running into a file handle shortage rather than a stream shortage. ```lsof``` output:

```
caed    25307 root   41r      REG               8,17   2373980 268507944 /var/snd/060001_011.wav
caed    25307 root   42r      REG               8,17  30159928 268504980 /var/snd/011588_001.wav
caed    25307 root   43r      REG               8,17   2123612 268484715 /var/snd/045003_010.wav
caed    25307 root   44r      REG               8,17  38969788 268504997 /var/snd/011593_001.wav
caed    25307 root   45r      REG               8,17   1501224 268481537 /var/snd/070000_004.wav
caed    25307 root   46r      REG               8,17  41027932 268485519 /var/snd/011333_001.wav
caed    25307 root   47r      REG               8,17   1555552 268507575 /var/snd/045001_021.wav
caed    25307 root   48r      REG               8,17   1338512 268507591 /var/snd/045001_026.wav
caed    25307 root   49r      REG               8,17   1738212 268507589 /var/snd/045001_025.wav
caed    25307 root   50r      REG               8,17   2193120 268484924 /var/snd/045001_024.wav
caed    25307 root   51r      REG               8,17   1841584 268484931 /var/snd/045001_029.wav
caed    25307 root   52r      REG               8,17   2853732 268507948 /var/snd/045001_031.wav
caed    25307 root   53r      REG               8,17  35445844 268506257 /var/snd/011503_001.wav
caed    25307 root   54r      REG               8,17   2565160 268507947 /var/snd/045001_030.wav
caed    25307 root   55r      REG               8,17  40639680 268506225 /var/snd/011239_001.wav
caed    25307 root   56r      REG               8,17  31234176 268507600 /var/snd/011645_001.wav
caed    25307 root   57r      REG               8,17    449500 268503852 /var/snd/045008_011.wav
caed    25307 root   58r      REG               8,17  40003880 268439908 /var/snd/010017_001.wav
caed    25307 root   59r      REG               8,17   2123612 268484715 /var/snd/045003_010.wav
caed    25307 root   60r      REG               8,17  54524876 268435974 /var/snd/011267_001.wav
caed    25307 root   61r      REG               8,17   1555552 268507575 /var/snd/045001_021.wav
caed    25307 root   62r      REG               8,17  35802252 268518742 /var/snd/011559_001.wav
caed    25307 root   63r      REG               8,17   1215392 268485184 /var/snd/045005_001.wav
caed    25307 root   64r      REG               8,17  47724428 268484964 /var/snd/010967_001.wav
caed    25307 root   65r      REG               8,17   1338512 268507591 /var/snd/045001_026.wav
caed    25307 root   66r      REG               8,17  41549996 268485000 /var/snd/011003_001.wav
caed    25307 root   67r      REG               8,17  45808516 268485534 /var/snd/011580_001.wav
caed    25307 root   68r      REG               8,17   1525820 268500768 /var/snd/045008_007.wav
caed    25307 root   69r      REG               8,17  35260108 268484782 /var/snd/010785_001.wav
caed    25307 root   70r      REG               8,17  29662960 268507611 /var/snd/011334_001.wav
caed    25307 root   71r      REG               8,17   6141920 268507593 /var/snd/045004_005.wav
caed    25307 root   72r      REG               8,17  11546816 268504993 /var/snd/060168_001.wav
caed    25307 root   73r      REG               8,17   5743308 268439801 /var/snd/060119_001.wav
caed    25307 root   74r      REG               8,17  11552704 268500894 /var/snd/060164_001.wav
caed    25307 root   75r      REG               8,17  11562248 268505015 /var/snd/060200_001.wav
caed    25307 root   76r      REG               8,17   3112896 268485161 /var/snd/045003_001.wav
caed    25307 root   77r      REG               8,17  32579500 268475376 /var/snd/010221_001.wav
caed    25307 root   78r      REG               8,17  39637428 268505306 /var/snd/011204_001.wav
caed    25307 root   79r      REG               8,17   1738212 268507589 /var/snd/045001_025.wav
caed    25307 root   80r      REG               8,17  37825260 268481317 /var/snd/010389_001.wav
caed    25307 root   81r      REG               8,17  43613164 268475459 /var/snd/010304_001.wav
caed    25307 root   82r      REG               8,17   3467592 268507946 /var/snd/060001_009.wav
caed    25307 root   83r      REG               8,17  37036024 268518748 /var/snd/011650_001.wav
caed    25307 root   84r      REG               8,17   2636012 268485165 /var/snd/045003_002.wav
caed    25307 root   85r      REG               8,17  40520924 268524869 /var/snd/011654_001.wav
caed    25307 root   86r      REG               8,17    804300 268485496 /var/snd/045008_005.wav
caed    25307 root   87r      REG               8,17  43387268 268484905 /var/snd/010908_001.wav
caed    25307 root   88r      REG               8,17  32418868 268518743 /var/snd/011560_001.wav
caed    25307 root   89r      REG               8,17   2193120 268484924 /var/snd/045001_024.wav
caed    25307 root   90r      REG               8,17  37995936 268504027 /var/snd/011487_001.wav
caed    25307 root   91r      REG               8,17   1903112 268485166 /var/snd/045003_003.wav
caed    25307 root   92r      REG               8,17  40508496 268507966 /var/snd/010850_001.wav
caed    25307 root   93r      REG               8,17   1841584 268484931 /var/snd/045001_029.wav
caed    25307 root   94r      REG               8,17  36740968 268484915 /var/snd/010918_001.wav
caed    25307 root   95r      REG               8,17   1215392 268485184 /var/snd/045005_001.wav
caed    25307 root   96r      REG               8,17  43042012 268507927 /var/snd/011620_001.wav
caed    25307 root   97r      REG               8,17   2853732 268507948 /var/snd/045001_031.wav
caed    25307 root   98r      REG               8,17  32395436 268505304 /var/snd/011599_001.wav
caed    25307 root   99r      REG               8,17  44356100 268503991 /var/snd/011594_001.wav
caed    25307 root  100r      REG               8,17   2565160 268507940 /var/snd/060001_010.wav
caed    25307 root  101r      REG               8,17  34982292 268507552 /var/snd/011186_001.wav
caed    25307 root  102r      REG               8,17   2565160 268507947 /var/snd/045001_030.wav
caed    25307 root  103r      REG               8,17   1555552 268507575 /var/snd/045001_021.wav
caed    25307 root  104r      REG               8,17  43447508 268439806 /var/snd/010441_001.wav
caed    25307 root  105r      REG               8,17   6141920 268507593 /var/snd/045004_005.wav
caed    25307 root  106r      REG               8,17  11541364 268500875 /var/snd/060157_001.wav
caed    25307 root  107r      REG               8,17   1566780 268485167 /var/snd/045003_004.wav
caed    25307 root  108r      REG               8,17  33727828 268981754 /var/snd/011564_001.wav
caed    25307 root  109r      REG               8,17   1338512 268507591 /var/snd/045001_026.wav
caed    25307 root  110r      REG               8,17  33322440 268484943 /var/snd/010945_001.wav
caed    25307 root  111r      REG               8,17  41319080 268484672 /var/snd/010675_001.wav
caed    25307 root  112r      REG               8,17  44351084 268484696 /var/snd/010699_001.wav
caed    25307 root  113r      REG               8,17   2373980 268507944 /var/snd/060001_011.wav
caed    25307 root  114r      REG               8,17  37515820 268507910 /var/snd/011574_001.wav
caed    25307 root  115r      REG               8,17   2154104 268485170 /var/snd/045003_005.wav
caed    25307 root  116r      REG               8,17  31249236 268484901 /var/snd/011307_001.wav
caed    25307 root  117r      REG               8,17   1808692 268507590 /var/snd/070000_003.wav
caed    25307 root  118r      REG               8,17  42192540 268475475 /var/snd/010320_001.wav
caed    25307 root  119r      REG               8,17  47953288 268518696 /var/snd/011644_001.wav
caed    25307 root  120r      REG               8,17   1738212 268507589 /var/snd/045001_025.wav
caed    25307 root  121r      REG               8,17  44456500 268505901 /var/snd/011496_001.wav
```

**Comment by deltecent** [comment]
This may be related to https://github.com/ElvishArtisan/rivendell/commit/b0e92a7899b9a22a9447189780d37a67bd991a4f#diff-7332e8aa1996813dc4ee8fd084d7b49c

**Comment by deltecent** [comment]
Fixed in #440
